libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -1626,7 +1628,7 @@ private static (object value, string error) Where(Expression expression, IMemory
 
                         // the local iterator is pushed as one memory layer in the memory stack
                         stackedMemory.Push(new SimpleObjectMemory(local));
-                        var (r, _) = expression.Children[2].TryEvaluate<bool>(stackedMemory);
+                        var (r, _) = expression.Children[2].TryEvaluate<bool>(stackedMemory, options);
```

Comment:
>options [](start = 93, length = 7)

Boolean context.

Reply:
great catch, i just search all isLogicTrue, forget here

Reply:
Actually, this should use isLogicTrue anyway.  If your test was a variable it won't have the right logic.

---
In reply to: [399385504](https://github.com/microsoft/botbuilder-dotnet/pull/3635#discussion_r399385504) [](ancestors = 399385504)